payone-adapter/src/main/java/com/commercetools/sunrise/payment/payone/PayonePaymentServiceProviderImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.commercetools.sunrise.payment.payone;
+
+import com.commercetools.sunrise.payment.domain.CreatePaymentDataProvider;
+import com.commercetools.sunrise.payment.domain.PaymentServiceProvider;
+import com.commercetools.sunrise.payment.utils.PaymentPropertiesLoadingHelper;
+import io.sphere.sdk.payments.Payment;
+import io.sphere.sdk.payments.PaymentMethodInfo;
+import io.sphere.sdk.payments.PaymentStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Created by mgatz on 7/18/16.
+ */
+public class PayonePaymentServiceProviderImpl implements PaymentServiceProvider {
+
+    private PaymentPropertiesLoadingHelper propertiesLoadingHelper;
+
+    public PayonePaymentServiceProviderImpl() {
+        propertiesLoadingHelper = PaymentPropertiesLoadingHelper.createFromResource("methods/payone.properties");
+    }
+
+    @Override
+    public String getId() {
+        return propertiesLoadingHelper.getProperty("methods.interface");
+    }
+
+    @Override
+    public List<PaymentMethodInfo> getAvailablePaymentMethods() {
+        return propertiesLoadingHelper.getAvaiableMethodIds().stream().map(id -> propertiesLoadingHelper.getMethodInfo(id)).collect(Collectors.toList());
+    }
+
+    @Override
+    public Function<CreatePaymentDataProvider, Payment> provideCreatePaymentHandler(String methodId) {
```

Comment:
Could you please give a use case why is it better to use function instead of simple ```Payment``` object as return value?